### PR TITLE
Added new fluentbit-metrics receiver

### PR DESCRIPTION
**Pull request by rafaelwestphal**
The receiver will be responsible for making the fluentbit internal
metrics available.

### CODE REVIEW

File: confgenerator/logging_receivers.go [hunk 1]
```diff
@@ -272,3 +272,28 @@ func (r LoggingReceiverSystemd) Components(tag string) []fluentbit.Component {
...
+type LoggingReceiverMetric struct {
+	ConfigComponent `yaml:",inline"`
+}
+
+func (r LoggingReceiverMetric) Components(tag string) []fluentbit.Component {
+	return []fluentbit.Component{{
+		Kind: "INPUT",
+		Config: map[string]string{
+			// https://docs.fluentbit.io/manual/pipeline/inputs/systemd
+			"Name":            "fluentbit_metrics",
```
Comment (qingling128) [inline confgenerator/logging_receivers.go:286]: Similar to before, we do not want to expose the implementation details of Fluent Bit to users. Instead, we make the self metrics enabled by default and users can not turn it off. This way we guarantee:

* Ops Agent self metrics are always ingested 
* In the future, we have flexibility to swap the underlying agents (e.g. Fluent Bit or Open Telemetry Collector) if needed.

Comment (rafaelwestphal) [reply confgenerator/logging_receivers.go]: Changed implementation to hardcode the data as discussed

File: confgenerator/fluentbit/service.go [hunk 2]
```diff
@@ -55,3 +55,15 @@ func (s Service) Component() Component {
 		},
 	}
 }
+
+func (s Service) MetricsComponent() Component {
+	return Component{
+		Kind: "INPUT",
+		Config: map[string]string{
+			// https://docs.fluentbit.io/manual/pipeline/inputs/systemd
```
Comment (qingling128) [inline confgenerator/fluentbit/service.go:63]: We probably wanna update this URL?

File: confgenerator/testdata/valid/linux/all-backward_compatible_with_explicit_exporters/golden_fluent_bit_main.conf [hunk 3]
```diff
@@ -15,6 +15,11 @@
     storage.metrics           on
     storage.sync              normal
 
+[INPUT]
+    Name            fluentbit_metrics
+    Scrape_Interval 30
```
Comment (qingling128) [inline confgenerator/testdata/valid/linux/all-backward_compatible_with_explicit_exporters/golden_fluent_bit_main.conf:20]: Ops Agent's default collection interval is 60s. We might wanna match that.

**Review by qingling128 — APPROVED** [review]
Two minor comments. Otherwise LGTM.

**Comment by qingling128** [comment]
As discussed offline, we'd like to bundle the lated PRs into one commit when merging to master to ensure things can be rolled back together if needed, and the change to remove port 2020 metrics and expose new metrics end up in the same agent release.

**Comment by qingling128** [comment]
Obsolete by https://github.com/GoogleCloudPlatform/ops-agent/pull/330

**Pull request closed without merging**